x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/PITAwareQueryClient.java

```diff
@@ -91,6 +97,23 @@ private void searchWithPIT(MultiSearchRequest search, ActionListener<MultiSearch
         }, listener));
     }
 
+    private void makeRequestPITCompatible(SearchRequest request) {
+        SearchSourceBuilder source = request.source();
+        // don't increase the keep alive
+        source.pointInTimeBuilder(new PointInTimeBuilder(pitId));
+        // move the indices from the search request to a index filter - see #63132
+        String[] indices = request.indices();
+        if (CollectionUtils.isEmpty(indices) == false) {
+            request.indices(Strings.EMPTY_ARRAY);
+            BoolQueryBuilder indexFilter = boolQuery().filter(termsQuery(GetResult._INDEX, indices));
```

Comment:
I think `IndexFieldMapper` can't yet query against data streams. So if a data stream is provided as indices then that will correct always yield no results.

I think the `SearchIndexNameMatcher` class needs to be modified in order to take into account data streams. The 
`matchesIndex()` method should invoke the overloaded version of `concreteIndexNames()` method that accepts `includeDataStreams` parameter and provide `true`.

Reply:
Should I fix that in a separate change? I think a dedicated test for this should exists as well.

Reply:
+1

Reply:
Thanks for looking @martijnvg

Reply:
I've opened: #63170